plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -99,47 +86,62 @@ def main():
             application_id=dict(required=False),
             changelog=dict(required=False),
             description=dict(required=False),
-            revision=dict(required=False),
+            revision=dict(required=True),
             user=dict(required=False),
-            appname=dict(required=False),
-            environment=dict(required=False),
-            validate_certs=dict(default=True, type='bool'),
         ),
-        required_one_of=[['app_name', 'application_id']],
-        supports_check_mode=True
```

Comment:
You are removing check mode support. Is that really necessary? If yes, it must be properly documented (breaking_changes).

Reply:
check_mode is return true, which can't be expected result while making actual call to NewRelic so it's much better to remove support for check mode.
I have updated the changelog for it.

Reply:
Since the module always returns with `changed=True` (if it doesn't fail), this would be the correct behavior. I don't think you should remove check mode support.